### PR TITLE
initial version

### PR DESCRIPTION
**Pull request by travi**
doesnt yet
* remove the travis badge (coming soon as [a remark plugin](https://github.com/form8ion/remark-remove-travis-ci-badge)
* configure packages to run on multiple versions of node (will be added soon in the actions scaffolder)
* include the renovate branch in the workflow config (still pondering this one, but close to some ideas)
* handle non-package project visibility (could use some help thinking through options for what would be best to check on this case)

### CODE REVIEW

File: src/actions.js [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+
+  const packageDetails = JSON.parse(await fs.readFile(`${projectRoot}/package.json`, 'utf-8'));
+  const {scripts, publishConfig} = packageDetails;
+
+  return scaffold({
+    projectRoot,
+    vcs,
+    tests: {...scripts['test:unit'] && {unit: true}},
+    visibility: determineVisibility(publishConfig),
+    projectType: determineProjectType(packageDetails)
```
Comment (trevtrich) [inline src/actions.js:25]: this stuff is pretty cool 👍

File: src/project-type.js [hunk 2]
```diff
@@ -0,0 +1,9 @@
+import {projectTypes} from '@form8ion/javascript-core';
+
+export default function ({private: isApplication, main, bin}) {
```
Comment (trevtrich) [inline src/project-type.js:3]: could see this fitting in `@form8ion/javascript-core` eventually

**Review by trevtrich — APPROVED** [review]
This is pretty sweet. Love the derivation of state from the `package.json`.

**Comment by travi** [comment]
> Love the derivation of state from the package.json.

Yeah, thankfully we're consistent enough to have pretty good info for packages. I'm still not sure what to base `visibility` on for applications/non-packages. I'm open to ideas there

**Pull request merged**